### ISSUE TITLE
Connection not recovered when crossing networkless area 

### ISSUE THREAD

**Issue by binou06** [teslamate-org/teslamate#2490]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Hello,

when passing through a networkless area (the car does not have any network for few minutes), Teslamate seams to give up connecting (after car network recovery of course).

If I load the iPhone Tesla app, Teslamate recovers.

See explanation on this map

![teslamate-bug](https://user-images.githubusercontent.com/4947617/155111016-7fb8c48a-373e-4ef1-b3bd-34243a5d2360.jpg)
 
For further investigation :
- Teslamate 1.26.1 (regular install --> not docker)
- This behaviour is since several days (work like a charm for months before). Upgraded cars to 2022.4.5.1 (approximatively on the same time issue occurs --> not sure)
- This behaviour is for both of my Tesla

Seems like if Tesla introduced a new state "lost connection" (or something like that) and Teslamate do not want to go further not to awake the car... Loading tesla app on the phone could bring back status to "teslamate known status" (pure speculation, let the pro give their ideas).

### Expected Behavior

Teslamate should connect on car network recover.

### Steps To Reproduce

1. Drive your Tesla until reaching a "networkless area" to trigger `Vehicle went offline while driving`
2. Re-enter a covered area --> Teslamate acts as if the Tesla is still offline
3. Open Tesla App on the iPhone --> Teslamate connect back

### Relevant log output

```shell
Got 2 vehicles so I tried to get rid of lines dealing with my other Tesla



févr. 22 07:24:46 tesla teslamate[22946]: 2022-02-22 07:24:46.369 [notice]     :alarm_handler: {:clear, TeslaMate.Vehicles.Vehicle_2_api_error}
févr. 22 08:34:47 tesla teslamate[22946]: 2022-02-22 08:34:47.096 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
févr. 22 08:34:47 tesla teslamate[22946]: 2022-02-22 08:34:47.096 car_id=2 [error] Error / :unknown
févr. 22 08:35:06 tesla teslamate[22946]: 2022-02-22 08:35:06.162 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
févr. 22 08:35:06 tesla teslamate[22946]: 2022-02-22 08:35:06.162 car_id=2 [error] Error / :unknown
févr. 22 08:35:25 tesla teslamate[22946]: 2022-02-22 08:35:25.232 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
févr. 22 08:35:25 tesla teslamate[22946]: 2022-02-22 08:35:25.232 car_id=2 [error] Error / :unknown
févr. 22 08:35:44 tesla teslamate[22946]: 2022-02-22 08:35:44.304 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
févr. 22 08:35:44 tesla teslamate[22946]: 2022-02-22 08:35:44.304 car_id=2 [error] Error / :unknown
févr. 22 08:35:46 tesla teslamate[22946]: 2022-02-22 08:35:46.441 [notice]     :alarm_handler: {:set, {TeslaMate.Vehicles.Vehicle_2_api_error, :fuse_blown}}
févr. 22 08:35:52 tesla teslamate[22946]: 2022-02-22 08:35:52.651 car_id=2 [info] Received stream data: :inactive
févr. 22 08:35:54 tesla teslamate[22946]: 2022-02-22 08:35:54.433 car_id=2 [warning] Vehicle went offline while driving
févr. 22 08:40:54 tesla teslamate[22946]: 2022-02-22 08:40:54.243 car_id=2 [info] Received stream data: :inactive
févr. 22 08:42:46 tesla teslamate[22946]: 2022-02-22 08:42:46.448 [notice]     :alarm_handler: {:clear, TeslaMate.Vehicles.Vehicle_2_api_error}
févr. 22 08:42:55 tesla teslamate[22946]: 2022-02-22 08:42:55.920 car_id=2 [info] Fetch already in progress ...
févr. 22 08:42:55 tesla teslamate[22946]: 2022-02-22 08:42:55.920 car_id=2 [info] Fetch already in progress ...
févr. 22 08:42:55 tesla teslamate[22946]: 2022-02-22 08:42:55.927 car_id=2 [info] Fetch already in progress ...
févr. 22 08:42:56 tesla teslamate[22946]: 2022-02-22 08:42:56.146 car_id=2 [info] Vehicle came back online after 9 min
févr. 22 08:46:18 tesla teslamate[22946]: 2022-02-22 08:46:18.493 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (156.655 ms)
févr. 22 08:46:18 tesla teslamate[22946]: 2022-02-22 08:46:18.983 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (478.631 ms)
févr. 22 08:46:18 tesla teslamate[22946]: 2022-02-22 08:46:18.994 [info] End of drive initiated by: %TeslaApi.Vehicle.State.Drive{gps_as_of: 1645515976, heading: 188, latitude: 43.607621, longitude: X.XXXXXXX, native_latitude: X.XXXXXXX, native_location_supported: 1, native_longitude: X.XXXXXXX, native_type: "wgs", power: 1, shift_state: "P", speed: 0, timestamp: 1645515977145}
févr. 22 08:46:18 tesla teslamate[22946]: 2022-02-22 08:46:18.994 car_id=2 [info] Driving / Ended / 47 km – 99 min
```

### Type of installation

Manual

### Version

v1.26.1

**Comment by binou06** [comment]
The log I post covers the map I gave. Not sure "Vehicle Error: Vehicle is offline" for the car (probably the other car which was asleep --> I see this occurs just after questioning tesla api for my other car... 

I think investigation is more on the lines :
```
févr. 22 08:35:46 tesla teslamate[22946]: 2022-02-22 08:35:46.441 [notice]     :alarm_handler: {:set, {TeslaMate.Vehicles.Vehicle_2_api_error, :fuse_blown}}
févr. 22 08:35:52 tesla teslamate[22946]: 2022-02-22 08:35:52.651 car_id=2 [info] Received stream data: :inactive
```

**Comment by Shadesh** [comment]
Same problem here.
Teslamate: 1.27.0-dev
Tesla: 2022.4.5.1

**Comment by JakobLichterfeld** [comment]
From my observations, this happens when the network changed from WLAN to LTE or vise versa. Workaround: open Tesla App once

**Comment by binou06** [comment]
For me, it was not dependant of switching between WLAN/LTE.

To be sure I wanted to wait couple of days but (I cross the zone everyday) there is nom more problem. Perhaps relative to bad Tesla server response unhandled by teslamate ? 

What do you advise me ? Should I close the issue (as other people report same issue ?)

**Comment by woyteck1** [comment]
Actually yes, I have also noticed this recently, but I assumed just connection drop on Tesla's servers side...

**Comment by JakobLichterfeld** [comment]
> However looking into the DB, was wondering what is suspend_min=21 and suspend_after_idle_min=15  - andf whether it has something to do with not reading the values... of drives.

From my understanding: That are the values used to let the vehicle sleep when NOT using streaming API (which is default in new versions).

**Comment by woyteck1** [comment]
I just had a 130mile drive which wasn't recorded.
I am wondering whether this is because of the new update to the car? Mine is 2022.4.5.3
Or maybe Tesla API changed somewhere...

Sent from Yahoo Mail on Android 
 
  On Thu, 3 Mar 2022 at 17:24, ***@***.***> wrote:   



However looking into the DB, was wondering what is suspend_min=21 and suspend_after_idle_min=15 - andf whether it has something to do with not reading the values... of drives.


From my understanding: That are the values used to let the vehicle sleep when NOT using streaming API (which is default in new versions).

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by JakobLichterfeld** [comment]
Mine was in 2022.4.5 buggy, 2022.4.5.3 seemed already fixed and 2022.4.5.5 is all good till now, even with switch from WLAN to LTE and vise versa.

**Comment by woyteck1** [comment]
Correction. It did record the data, but only opened the drive and did not close it. Until the full stop/park I did.
However I did start the drive from home, while on Wi-Fi. 
Had the same thing happening on the way back, but it was because, while in EuroTunnel shuttle, I had to restart my infotainment (scary stuff, but Cuphead froze :/ ), then the car was not on any network, and it registered on Mobile network once we got out of the tunnel. 
So I had to fix two drives in the DB today. (I had a problem with that so will raise another issue because I know a fix as well).
